notebooks/WFC3/exception_report/wfc3_exception_report.ipynb

```diff
@@ -116,18 +116,21 @@
    "source": [
```

Comment:
<p>Thanks for your work. I have a suggestion regarding the file paths. Replacing string concatenation with <code>os.path.join</code> would be more robust and platform-independent way to handle file paths.</p><p>I also suggest providing feedback to user instead of silently passing. </p><p> </p><p>I had this when working you can test and try out if this suits:</p><p>Make sure to import shutils also.</p><p> </p><pre class="ql-syntax" spellcheck="false"># Loop through exposure id
for obsid in exp_ids:&nbsp;
&nbsp; &nbsp; # make new directory to hold fits files - named by exposure id
&nbsp; &nbsp; newdir = os.path.join(os.getcwd(), obsid.lower())
&nbsp; &nbsp; try:
&nbsp; &nbsp; &nbsp; &nbsp; mkdir = os.mkdir(newdir)
&nbsp; &nbsp; &nbsp; &nbsp; print(f'Making new directory {newdir}')
&nbsp; &nbsp; except FileExistsError:&nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; print(f'Directory {newdir} already exists.')
&nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; # Loop through to get FLTs, JIFs, and JITs
&nbsp; &nbsp; for file_type in file_types:
&nbsp; &nbsp; &nbsp; &nbsp; print(f'Working on getting {file_type} files for Exposure ID {obsid}')
&nbsp; &nbsp; &nbsp; &nbsp; obs_table = Observations.query_criteria(obs_id=obsid.lower())
&nbsp; &nbsp; &nbsp; &nbsp; products = Observations.get_product_list(obs_table)
&nbsp; &nbsp; &nbsp; &nbsp; filtered_products = Observations.filter_products(products, productSubGroupDescription=file_type, project='CALWF3')
&nbsp; &nbsp; &nbsp; &nbsp; download_table = Observations.download_products(filtered_products, mrp_only=False)
&nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; # For convenience move raws to cwd and remove empty download dir
&nbsp; &nbsp; &nbsp; &nbsp; for file in download_table['Local Path']:
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; filename = os.path.basename(file)
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; new_file_path = os.path.join(newdir, filename)
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; print(f'Moving {file} to {new_file_path}')
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; os.rename(file, new_file_path)

&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; remove_dir = os.path.join('mastDownload', 'HST', filename[:9])&nbsp; &nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; try:
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; os.rmdir(remove_dir)
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; print(f'Removing {remove_dir}')
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; except (OSError, FileNotFoundError):&nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; print(f'Error removing directory {remove_dir}')
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; mast_dir = 'mastDownload'
&nbsp; &nbsp; # Check and remove mastDownload directory
&nbsp; &nbsp; if os.path.exists(mast_dir):
&nbsp; &nbsp; &nbsp; &nbsp; print(f'Removing {mast_dir} directory')
&nbsp; &nbsp; &nbsp; &nbsp; shutil.rmtree(mast_dir)
&nbsp; &nbsp; else:
&nbsp; &nbsp; &nbsp; &nbsp; print(f'{mast_dir} does not exist') 
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/hst_notebooks/pull/102/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>